util/base/src/test/java/jetbrains/jetpad/base/edt/TestEventDispatchThread.java

```diff
@@ -101,20 +101,25 @@ private int executeCurrentUpdates() {
     int mc;
     do {
       mc = myModificationCount;
-      List<RunnableRecord> current = getCurrentRecords();
-      run(current);
-      runCommandsNum += current.size();
-      myRecords.removeAll(current);
+      runCommandsNum += runCurrentRecords();
     } while (myModificationCount != mc);
     return runCommandsNum;
   }
 
-  private void run(List<RunnableRecord> current) {
+  private int runCurrentRecords() {
+    List<RunnableRecord> current = getCurrentRecords();
+    int executedNum = 0;
     myRunning = true;
-    for (RunnableRecord r : current) {
-      r.myRunnable.run();
+    try {
+      for (RunnableRecord r : current) {
+        executedNum++;
+        r.myRunnable.run();
+      }
+    } finally {
+      myRunning = false;
```

Comment:
Since this edt was designed for tests, I see no reason for using ThrowableHandlers here.